config/opal_check_pmi.m4

```diff
@@ -39,18 +39,30 @@ AC_DEFUN([OPAL_CHECK_PMI_LIB],
     opal_check_$3_mycppflags=
 
     # check for the header
-    AC_MSG_CHECKING([for $3.h in $1/include])
-    AS_IF([test -f $1/include/$3.h],
-          [AC_MSG_RESULT([found])
-           opal_check_$3_mycppflags="-I$1/include"],
-          [AC_MSG_RESULT([not found])
-           AC_MSG_CHECKING([for $3.h in $1/include/slurm])
-           AS_IF([test -f $1/include/slurm/$3.h],
+    AS_IF([test -n "$1"],
```

Comment:
Urrr...what does ```test -n``` do?

Reply:
`test -n string` is equivalent to `test ! -z string`. In this case, `$1` is empty when configure'd with `--with-pmix` or `--with-pmix=yes`, and not empty when configure'd with `--with-pmix=DIR`